game_states.py

```diff
@@ -0,0 +1,114 @@
+import pymlgame
+import time
+
+import drawing
+from misc import Point
+
+
+class GameState():
+    """
+    Abstract base class for a state of a pyngpong game. Derived classes should
+    implement update() and render().
+    States will go ahead and replace themselves by another state if they deem
+    it necessary.
+    """
+
+    def __init__(self, game):
+        self.game = game
+
+    def update(self):
+        pass
+
+    def render(self):
+        pass
+
+
+class WaitingState(GameState):
+    """
+    In this state, the game waits for 2 players to join.
+    """
+
+    def update(self):
+        if len(self.game.players) == 2:
+            self.game.state = StartingState(self.game)
+
+    def render(self):
+        p0_color = pymlgame.GREEN if len(self.game.players) > 0 \
+            else pymlgame.GREY5
+        p1_color = pymlgame.GREEN if len(self.game.players) > 1 \
+            else pymlgame.GREY5
+
+        surface = pymlgame.Surface(
+            self.game.screen.width, self.game.screen.height)
+        # note(hrantzsch): this works for our default screen size only
+        drawing.draw_p(surface, Point(0, 2), p0_color)
+        drawing.draw_0(surface, Point(4, 2), p0_color)
+        drawing.draw_p(surface, Point(10, 2), p1_color)
+        drawing.draw_1(surface, Point(14, 2), p1_color)
+        self.game.screen.blit(surface)
+
+
+class StartingState(GameState):
+    """
+    In this state, the game renders only the players' paddles for a given
+    amount of time.
+    """
+
+    def __init__(self, game):
+        super(self.__class__, self).__init__(game)
+        self.init_time = time.time()
+        self.delay = 3  # seconds between inner state changes
```

Comment:
yeah, we could go for a config file. I do like [configparser](https://docs.python.org/3/library/configparser.html?highlight=config#module-configparser). However, I think this is too much overhead at this stage.